MerlinAU.sh

```diff
@@ -405,6 +406,38 @@ _ScriptVersionStrToNum_()
    echo "$verNum" ; return 0
 }
 
+##---------------------------------------##
+## Added by ExtremeFiretop [2024-Jun-19] ##
+##---------------------------------------##
+_GetFirmwareVariantFromRouter_()
+{
+   ##DEFAULTS TO MERLIN##
+   local retCode=0  newVersionStr
+
+   buildInfoStr="$(nvram get buildinfo)"
+   innerverStr="$(nvram get innerver)"
+
+   ##FOR TESTING/DEBUG ONLY##
+   if false # Change to true for forcing GNUton flag
+   then 
+      isGNUtonFW=true
+   else
+      # Check if innerver contains "gnuton"
+      if echo "$innerverStr" | grep -iq "gnuton"
```

Comment:
IIRC, the NVRAM "**extendno**" var also contains "**gnuton**" to indicate a Gnuton F/W, correct?
If yes, I think that it could be used in conjunction with the "**innerver**" var to double-check.
Something like this:

```
    if echo "$innerverStr" | grep -iq "gnuton" || \
       echo "$fwInstalledExtendNum" | grep -iq "gnuton"
    then
        isGNUtonFW=true
   ....
```

Reply:
Correct! And this is a good idea, i like the method. I didn't think about it at the time.
Will adjust accordingly ASAP!

Reply:
Done in commit: https://github.com/ExtremeFiretop/MerlinAutoUpdate-Router/pull/186/commits/721a7a94d7fdafb7238185fbee479ebbbc5dcbda

And yes I removed the debug afterwards, oopies :P